### ISSUE TITLE
Make `wrap` and friends generic over body type 

### ISSUE THREAD

**Issue by jtscuba** [paperclip-rs/paperclip#456]
The current definition for [wrap](https://github.com/paperclip-rs/paperclip/blob/master/plugins/actix-web/src/web.rs#L384) on web::Scope doesn't specify a type parameter for the response body, forcing it to use the default BoxBody. This makes it incompatible with middlewares that don't return a box body.

```
    pub fn wrap<M>(
        mut self,
        mw: M,
    ) -> Scope<
        actix_web::Scope<
            impl ServiceFactory<
                ServiceRequest,
                Config = (),
                Response = ServiceResponse,
                Error = Error,
                InitError = (),
            >,
        >,
    >
    where
        M: Transform<
                T::Service,
                ServiceRequest,
                Response = ServiceResponse,
                Error = Error,
                InitError = (),
            > + 'static,
```

Conversely the [wrap impl](https://github.com/paperclip-rs/paperclip/blob/master/plugins/actix-web/src/web.rs#L156) for service does specify this parameter
```
pub fn wrap<M, B>(
        self,
        mw: M,
    ) -> Resource<
        actix_web::Resource<
            impl ServiceFactory<
                ServiceRequest,
                Config = (),
                Response = ServiceResponse<B>,
                Error = Error,
                InitError = (),
            >,
        >,
    >
    where
        B: MessageBody,
        M: Transform<
                T::Service,
                ServiceRequest,
                Response = ServiceResponse<B>,
                Error = Error,
                InitError = (),
            > + 'static,
```
It looks like this extra parameter is missing from both scope and service response in the [web3](https://github.com/paperclip-rs/paperclip/blob/master/plugins/actix-web/src/web3.rs) version.

Is there a technical limitation to adding these parameters? It seems relatively straightforward to me.

**Comment by tiagolobocastro** [comment]
hmm not that I know of @jtscuba, but I haven't looked at that for a long time..
In other words, if you'd like to work on this a PR would be most welcome :)

**Comment by tiagolobocastro** [comment]
Closed by #457
